tools/update.php

```diff
@@ -0,0 +1,746 @@
+        . "is $release_version. "
+        . PHP_EOL
+        . 'Please run this script with the --confirm flag '
+        . 'if you wish to update your environment and source code to be '
+        . "compatiable with version $release_version."
+        . PHP_EOL;
+    die(usageString());
+}
+// Display some info and wait so the user can read it.
+echo 'This script will prompt for superuser privileges,'
+    . ' as they are needed to e.g. update apt packages.' . PHP_EOL;
+echo 'You may wish to review code changes tagged with "Caveat For '
+    . 'Existing Projects" as they may include changes to your workflow. '
+    . PHP_EOL
+    . "\t" . 'See: https://github.com/aces/Loris/pulls?'
+    . 'q=is%3Apr+label%3A%22Caveat+for+Existing+Projects%22+is%3Amerged'
+    . PHP_EOL . PHP_EOL;
+sleep(3);
+
+/* BEGIN UPDATE PROCESS */
+
+/* Below are all the apt packages required for LORIS to run.
+ *      @see: https://github.com/aces/Loris/wiki/Installing-Loris-in-Depth
+ */
+$loris_requirements = [
+                       "wget",
+                       "zip",
+                       "unzip",
+                       "php-json",
+                       "npm",
+                       "software-properties-common",
+                       "php-ast",
+                       "php$php_version",
+                       "php$php_version-mysql",
+                       "php$php_version-xml",
+                       "php$php_version-json",
+                       "php$php_version-mbstring",
+                       "php$php_version-gd",
+                       "libapache2-mod-php$php_version",
+                      ];
+
+// Update source code.
+echo '[***] Beginning LORIS update process.' . PHP_EOL;
+echo '[*] Release notes:' . PHP_EOL;
+echo $info->{'body'} . PHP_EOL . PHP_EOL;
+echo "[**] Updating LORIS source code "
+    . "($preupdate_version --> $release_version)" . PHP_EOL;
+if (updateSourceCode($loris_root_dir, $backup_dir)) {
+    echo '[+] LORIS source code files successfully updated.' . PHP_EOL;
+}
+
+// Update apt packages.
+echo '[**] Updating required packages...' . PHP_EOL;
+if (updateRequiredPackages($loris_requirements)) {
+    echo '[**] Required apt packages up-to-date.' . PHP_EOL;
+}
+
+// Update other dependencies via e.g. composer and npm
+echo '[**] Updating dependencies via package managers...' . PHP_EOL;
+chdir($loris_root_dir); // Composer will fail if not in LORIS root
+if (runPackageManagers()) {
+    echo '[**] Dependencies up-to-date.' . PHP_EOL;
+}
+
+// Print required SQL patches and commands needed to apply them
+$patches = getPatchesFromVersion(
+    $loris_root_dir,
+    $preupdate_version,
+    $release_version
+);
+if ($patches) {
+    echo "[*] Patches to update:" . PHP_EOL;
+    foreach ($patches as $filename) {
+        echo "\t] " . $filename . PHP_EOL;
+    }
+    echo '[*] Applying SQL patches...' . PHP_EOL;
+    applyPatches($patches, $db_config, $apply_patches);
+}
```

Comment:
shouldn't we add the `make` step from @driusan 's changes ?
https://github.com/aces/Loris/pull/3891

Reply:
Well this is targeted to the minor release, not the major release. But otherwise yes.

Reply:
dont we still need to recompile even between minor releases @driusan ?

Reply:
The make file doesn't exist in the minor version, I mean

Reply:
ohh, that makes total sense

:man_facepalming: